autopush/router/gcm.py

```diff
@@ -53,14 +58,33 @@ def route_notification(self, notification, uaid_data):
 
     def _route(self, notification, router_data):
         """Blocking GCM call to route the notification"""
+        data = {"chid": notification.channel_id,
+                "ver": notification.version}
+        # Payload data is optional.  If present, all of Content-Encoding,
+        # Encryption, and Encryption-Key are required.  If one or more are
+        # missing, a 400 response is produced.
+        con = notification.headers.get('content-encoding', None)
+        enc = notification.headers.get('encryption', None)
+        enckey = notification.headers.get('encryption-key', None)
```

Comment:
No need to search for them if there's no notification data, right? So they could go under the if statement below?